src/lib/derive-key-event-state.ts

```diff
@@ -0,0 +1,38 @@
+import { KeyboardEvent } from 'react'
+
+/**
+ * Returns an object with several booleans. The booleans describe the state of
+ * the `keyboardEvent` passed in as the single argument to this function. The
+ * booleans returned can be used to add custom keyboard interaction to an
+ * element within a `keydown` or `keypress` listener.
+ */
+function deriveKeyEventState(keyboardEvent: KeyboardEvent) {
+  const { key, shiftKey } = keyboardEvent
+
+  // Arrow keys
+  const isArrowDown = key === 'ArrowDown'
+  const isArrowLeft = key === 'ArrowLeft'
+  const isArrowRight = key === 'ArrowRight'
+  const isArrowUp = key === 'ArrowUp'
+
+  // Escape, Space, and Enter
+  const isEscape = key === 'Escape'
+  const isSpace = key === ' '
+
```

Comment:
TODO: add section here for Home, End, Page Up, and Page Down

Reply:
Done in https://github.com/hashicorp/dev-portal/pull/241/commits/eeac712df1c217d6479b4f6deabdd0862041f886